code/obj/machinery/manufacturer.dm

```diff
@@ -1026,6 +1039,22 @@
 					user.u_equip(W)
 					W.dropped()
 
+		else if (istype(W,/obj/item/disk/data/floppy/read_only/manudrive))
+			if (src.manudrive)
+				boutput(user, "<span class='alert'>There's already a drive in the machine. You need to remove it first.</span>")
+			else
+				boutput(user, "<span class='notice'>You insert [W].</span>")
+				W.set_loc(src)
+				src.manudrive = W
+				for (var/datum/computer/file/manudrive/MD in src.manudrive.root.contents)
+					src.drive_recipes = MD.drivestored
+					if (user && W)
+						user.u_equip(W)
+						W.dropped()
```

Comment:
Uhh, this should be before the set_loc I think but most definitely outside of the loop.

Reply:
Done! No clue why it was like that so uhh yeah thats fun.